### PR TITLE
[FLINK-3209] Remove Unused ProcessingTime, EventTime and AbstractTime

### PR DESCRIPTION
**Pull request by aljoscha**
Only keep Time for specifying time durations/intervals.

### CODE REVIEW

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/FoldWindowFunctionTest.java [hunk 1]
```diff
@@ -66,10 +72,27 @@ public Integer fold(Integer accumulator, Integer value) throws Exception {
 			}
 		);
 
-		AccumulatingProcessingTimeWindowOperator<Integer, Integer, Integer> windowOperator = new AccumulatingProcessingTimeWindowOperator<>(
-			foldWindowFunction,
+//		AccumulatingProcessingTimeWindowOperator<Integer, Integer, Integer> windowOperator = new AccumulatingProcessingTimeWindowOperator<>(
```
Comment (rmetzger) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/FoldWindowFunctionTest.java]: why is this commented out?

Comment (aljoscha) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/FoldWindowFunctionTest.java]: Must be an artifact from something, no idea. I'm removing that change.

**Comment by fhueske** [comment]
LGTM

**Comment by fhueske** [comment]
Will merge this PR.

**Pull request closed without merging**